packages/admin-next/dashboard/src/routes/orders/returns-create/components/create-returns-form/create-returns-form.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useState } from "react"
+import { useForm } from "react-hook-form"
+
+import { Order } from "@medusajs/medusa"
+import { Button, ProgressStatus, ProgressTabs } from "@medusajs/ui"
+import { useTranslation } from "react-i18next"
+import * as zod from "zod"
+
+import { RouteFocusModal } from "../../../../../components/route-modal"
+import { ItemsTable } from "../items-table"
+import { ReturnsForm } from "./returns-form"
+
+type CreateReturnsFormProps = {
+  order: Order
+}
+
+enum Tab {
+  ITEMS = "items",
+  DETAILS = "details",
+}
+
+type StepStatus = {
+  [key in Tab]: ProgressStatus
+}
+
+const CreateReturnSchema = zod.object({
+  quantity: zod.record<string, number>(zod.string(), zod.number()),
+  location: zod.string(),
+  shipping: zod.string(),
+  send_notification: zod.boolean().optional(),
+
+  enable_custom_refund: zod.boolean().optional(),
+  enable_custom_shipping_price: zod.boolean().optional(),
+
+  custom_refund: zod.number().optional(),
+  custom_shipping_price: zod.number().optional(),
+})
+
+export function CreateReturnsForm({ order }: CreateReturnsFormProps) {
+  const { t } = useTranslation()
+
+  const [selectedItems, setSelectedItems] = useState([])
+  const [tab, setTab] = React.useState<Tab>(Tab.ITEMS)
+
+  const isSubmitting = false
+  const isEditDirty = false
+
+  const selected = order.items.filter((i) => selectedItems.includes(i.id))
+
+  const form = useForm<typeof CreateReturnSchema>({
+    defaultValues: {
+      quantity: {},
+    },
```

Comment:
suggestion: make sure to initialize a default value for every value in a form. Otherwise, React will throw warnings for the fields because the change from being unmanaged to managed, and the form will initialize as being dirty, meaning that the user will see the prompt that they have unsaved changes even if they haven't touched the form.